libsplinter/src/admin/service/event/store/memory.rs

```diff
@@ -0,0 +1,319 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Defines a memory-backed implementation of the `AdminServiceEventStore`.
+//!
+//! The public interface includes the struct [`MemoryAdminServiceEventStore`].
+//!
+//! [`MemoryAdminServiceEventStore`]: struct.MemoryAdminServiceEventStore.html
+
+use std::collections::BTreeSet;
+use std::convert::TryInto;
+use std::sync::{Arc, Mutex};
+use std::time::SystemTime;
+
+use crate::admin::service::event::store::{AdminServiceEventStore, AdminServiceEventStoreError};
+use crate::admin::service::mailbox::EventEntry;
+use crate::admin::service::messages::AdminServiceEvent;
+use crate::error::InternalError;
+
+#[derive(Default, Clone)]
+pub struct MemoryAdminServiceEventStore {
```

Comment:
Can you add documentation comments for MemoryAdminServiceEventStore

Reply:
Updated!